### ISSUE TITLE
Typescript definition: disable allowSyntheticDefaultImports behavior

### ISSUE THREAD

**Issue by rokoroku** [async-library/react-async#109]
```ts
import React from "react"
       ~~~~~
// .../node_modules/@types/react/index"' can only be default-imported using the 'allowSyntheticDefaultImports' flag
```

Importing `react-async` from non-`allowSyntheticDefaultImports`-enabled project, the included typescript definition emits above error message. 

Since **the default value of the flag is not true**, it'd be better turning it off I think.
https://github.com/Microsoft/TypeScript/issues/10895

**Comment by ghengeveld** [comment]
I'm not familiar with this particular setting or situation, so I'm not sure how to resolve this. What do you suggest we do to fix this? Perhaps you can open a PR to fix our setup?

**Comment by rokoroku** [comment]
Sure, I'll submit a simple PR :)

**Comment by rokoroku** [comment]
Fixed by #112
